src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java

```diff
@@ -0,0 +1,195 @@
+          writer.value(TIME_FORMATTER.format(
+            Objects.requireNonNull(LocalTime.ofNanoOfDay(TimeUnit.MILLISECONDS.toNanos(((Integer) object))))));
+          break;
+        case TIME_MICROS:
+          writer.value(TIME_FORMATTER.format(
+            Objects.requireNonNull(LocalTime.ofNanoOfDay(TimeUnit.MICROSECONDS.toNanos((Long) object)))));
+          break;
+        case TIMESTAMP_MILLIS:
+          //timestamp for json input should be in this format yyyy-MM-dd HH:mm:ss.SSSSSS
+          writer.value(DATETIME_FORMATTER.format(
+            Objects.requireNonNull(getZonedDateTime((long) object, TimeUnit.MILLISECONDS))));
+          break;
+        case TIMESTAMP_MICROS:
+          writer.value(DATETIME_FORMATTER.format(
+            Objects.requireNonNull(getZonedDateTime((long) object, TimeUnit.MICROSECONDS))));
+          break;
+        default:
+          throw new IllegalStateException(
+            String.format("Field '%s' is of unsupported type '%s'", name, logicalType.getToken()));
+      }
+      return;
+    }
+
+    switch (schema.getType()) {
+      case NULL:
+        writer.nullValue(); // nothing much to do here.
+        break;
+      case INT:
+      case LONG:
+      case FLOAT:
+      case DOUBLE:
+        writer.value((Number) object);
+        break;
+      case BOOLEAN:
+        writer.value((Boolean) object);
+        break;
+      case STRING:
+        writer.value(object.toString());
+        break;
+      // TODO CDAP-15342 write byte type to json writer
+      default:
+        throw new IllegalStateException(String.format("Field '%s' is of unsupported type '%s'",
+                                                      name, schema.getType()));
+    }
+  }
+
+  private static void writeArray(JsonWriter writer, String name, Object value, Schema fieldSchema) throws IOException {
+    if (value == null) {
+      throw new RuntimeException(
+        String.format("Field '%s' is of value null, which is not a valid value for BigQuery type array.", name));
+    }
+
+    if (!(value instanceof Collection) && !value.getClass().isArray()) {
+      throw new IllegalArgumentException("The value should be of type collection or array. Got: " + value.getClass());
+    }
+
+    Schema componentSchema = BigQueryUtil.getNonNullableSchema(fieldSchema.getComponentSchema());
+    if (BigQueryUtil.UNSUPPORTED_ARRAY_TYPES.contains(componentSchema.getType())) {
+      throw new IllegalArgumentException(String.format("Field '%s' is of %s type within array, " +
+                                                         "which is not a valid BigQuery type.",
+                                                       name, componentSchema));
+    }
+
+    writer.name(name);
+    writer.beginArray();
+    if (value instanceof Collection) {
+      for (Object element : (Collection) value) {
+        // big query does not allow null values in array items
+        if (element == null) {
+          throw new IllegalArgumentException("");
```

Comment:
should have an error message, something like:

Field '%s' contains null values in its array, which is not allowed by BigQuery.